### ISSUE TITLE
DisassociateInfo should add document for custom implementing akka-remote-transport

### ISSUE THREAD

**Issue by hepin1989** [akka/akka#16374]
Hi currently the code here
https://github.com/akka/akka/blob/release-2.3-dev/akka-remote/src/main/scala/akka/remote/transport/Transport.scala#L172
no add anything about ```when to return what```
this is a little confusing 
and when looking inside the akka's netty transport implementation.
https://github.com/akka/akka/blob/release-2.3-dev/akka-remote/src/main/scala/akka/remote/transport/netty/TcpSupport.scala#L41
it return ```UNKNOWN``` but nothing more about this.
This should be clarified

**Comment by hepin1989** [comment]
I come to this again.
I have some question about this.
```
  /**
   * Supertype of possible disassociation reasons
   */
  sealed trait DisassociateInfo

  case object Unknown extends DisassociateInfo
  case object Shutdown extends DisassociateInfo
  case object Quarantined extends DisassociateInfo
```

what info should be notified when received 
```
PeerClosed,
ErrorClosed,
Closed
```
when implement the transport via akka-io tcp?

**Comment by drewhk** [comment]
So, these information tags are used by the high-level Remoting entry point which is not aware of any transport related specifics. It does not care if it is TCP or SCTP or something else. 

 - Shutdown means that the connection has been closed because the other system is shutting down. This is added by Akka itself in higher layers, you should not use this in a transport layer. 
 - Quarantined means that the connection has been closed because this system is quarantined on the other end of the connection. Again, this is added by Akka in higher layers and transports don't need to deal with these

Basically the only thing a transport can say is Unknown, and higher layers might add additional information if they can.

**Comment by hepin1989** [comment]
@drewhk thanks.
akka is moving to the stream based transport,and then will may comes with so many transport implementations too,so I think the an enhancement of the doc of the design will helps a lot.

I think what you just commented here is worth to add to the current doc of release 2.3.
I will try ,cause I just implemented one. thanks again.
